src/query/service/src/pipelines/processors/transforms/transform_resort_addon_without_source_schema.rs

```diff
@@ -63,13 +80,16 @@ pub fn build_expression_transform(
                 // but for pg or snowflake, if a field is non-null, it will return
                 // a non-null error (it means they will give a null as the default value).
                 if !f.is_nullable() {
-                    // if we have a user-specified default expr, it must satisfy the non-null constraint
-                    // in table-create phase. So we just consider default_expr is none.
-                    return Err(ErrorCode::BadArguments(format!(
-                        "null value in column `{}` of table `{}` violates not-null constraint",
-                        f.name(),
-                        table.name()
-                    )));
+                    *trigger_non_null_error = (true, f.name().to_string());
```

Comment:
I see. But we can improve the way we implement it by simply returning the error here. Change in `TransformResortAddOnWithoutSourceSchema: :try_create` function to judge whether the `build_expression_transform` is an error, in this way the code is more simple and easy to understand.
for example:
```rust
match build_expression_transform(
    input_schema,
    output_schema.clone(),
    table.clone(),
    ctx.clone(),
) {
    Ok(expression_transform) => {
        expression_transforms.push(expression_transform);
    }
    Err(err) => {
        errors(err);
    }
}
```

Reply:
So I need to check the error type, because we could receive other errors, not only the `non-null constraint` error, so do you think this way will be better? cc @b41sh

Reply:
So I need to check the error type, because we could receive other errors, not only the `non-null constraint` error, so do you think this way will be better? cc @b41sh

Reply:
It seems we don't need to check the error type. If t1 matched all data from t2, actually we don't need to run `build_expression_transform`. And if t1 does not match all data from t2, we can return any type of error at transform phase.

Reply:
No, we can't make sure whether t1 can matched all t2's data in pipeline build phase, the `build_expression_transform` is invoked in `try_create`, not `processor's porcess`